### ISSUE TITLE
PR diff LHS commit is reported as missing

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode-pull-request-github#3941]
1. Open https://github.com/microsoft/vscode-pull-request-github/pull/3936/ in a Codespace
2. Open the file diff 
3. :bug: diff fails to open and reports that the LHS commit doesn't exist, but from `git log` the commit does exist

![image](https://user-images.githubusercontent.com/30305945/189942104-40cd3808-e0af-46c7-8402-67944f46e402.png)

**Comment by joyceerhl** [comment]
Also, clicking on the active comment opens an empty file:

![image](https://user-images.githubusercontent.com/30305945/189942915-3e5fbf7d-838c-425d-a66a-0f3c8b8f5c0f.png)

**Comment by alexr00** [comment]
I could repro the first time I connected to the Codespace, but not the second. The first time, there `git log` was incomplete. Example from another PR since I can no longer repro with the original:

```
@alexr00 ➜ /workspaces/vscode-pull-request-github (alexr00/frightened-tick) $ git log
commit 937e2a0fc268dd38eebfb080698785ec2d3bc43a (grafted, HEAD -> alexr00/frightened-tick, origin/alexr00/frightened-tick)
Author: Alex Ross <alros@microsoft.com>
Date:   Wed Sep 14 11:56:34 2022 +0200

    WIP for cycle fix
@alexr00 ➜ /workspaces/vscode-pull-request-github (alexr00/frightened-tick) $ 
```

After I re-connect to the Codespace the `git log` is complete and the diffs work as expected. I assume Codespaces is grafting to speed up the Codespace creation in some way. I'll see if there's something I can listen for to know when the full history is ready.

**Comment by alexr00** [comment]
Did some testing, and it looks like it takes about 2 minutes before we have full git commit history. Even if there's an event we can wait on, that's too long to wait. I'm reaching out to the Codespaces folks to see if that can be improved.

**Comment by alexr00** [comment]
Heard back from the Codespaces folks. The delay in getting the full commit history is by design and can't be sped up.

Since waiting 2 minutes for the repo to be unshallowed isn't a good option, the only other thing I can think of is to get the LHS from github.com as a fallback.
